File: abelkhan_rpc_demo/proto/hub_call_client/csharp/module/hcallcmodule.cs
/*this imp file is codegen by abelkhan for c#*/

using System;
using System.Collections;
using System.Collections.Generic;

using common;

namespace imp
{
    public class hcallc_module : common.imodule
    {
        public string module_name;

        public delegate void hcallc_handle(String argv0);
        public event hcallc_handle onhcallc;
        void hcallc(ArrayList _event)
        {
            if (onhcallc == null)
            {
                return;
            }

            var argv0 = ((String)_event[0]);

            onhcallc(argv0);
        }

        public hcallc_module(client.client _client)
        {
            module_name = "hcallc";

            reg_event("hcallc", hcallc);

            _client.modulemanager.add_module("hcallc", this);
        }
    }
}

File: abelkhan_rpc_demo/proto/hub_call_client/js/module/hcallcmodule.js
/*this imp file is codegen by abelkhan for js*/
function hcallc(_client){
    event_cb.call(this);

    this.client_handle = _client;
    _client.modules.add_module("hcallc", this);

    this.hcallc = function(argv0)
    {
        this.call_event("hcallc", [argv0]);
    }

}
